builder/azure/arm/step_deploy_template.go

```diff
@@ -58,6 +58,18 @@ func (s *StepDeployTemplate) deployTemplate(ctx context.Context, resourceGroupNa
 	return err
 }
 
+func (s *StepDeployTemplate) deleteTemplate(ctx context.Context, resourceGroupName string, deploymentName string) error {
```

Comment:
I would call this `deleteDeployment`

Reply:
This matches the pattern set by the function above it that creates the Template/deploy, but I thought the same thing.

Reply:
maybe `deleteTemplateDeployment` then? The problem I have with `deleteTemplate` is that the "Template" is just the json representation that is used to create the actual "Deployment" resource. You're deleting the resource, not the json file/string.

Reply:
Cleanup always runs, regardless of fail/success. The only time it would not run is if packer literally crashes.

Reply:
Huh. Maybe I commented in the wrong place... I could have sworn there was some comment on here somewhere about someone wondering whether cleanup always ran or just when there was an error.